### ISSUE TITLE
Plugins won't load compiled object file on nix using ghc 7.8.4.

### ISSUE THREAD

**Issue by jraygauthier** [jgm/gitit#483]
> gitit: <interactive session>:
>    cannot find normal object file ‘wikistatic/private/plugins/GnuPlot.dyn_o’
 >   while linking an interpreted expression
> Gitit server closed.

Was working perfectly on nix using ghc 7.6.x.

Now, here is the situation, after the plugin load stage, I can find a `*.o` and a
`.hi` alongside the plugin's `*.hs`. However, the message seems to refer to
a missing `*.dyn_o`.

To me, it look like a problem with the recent use of dynamic libraries for ghc
packages. It may be a GHC issue, however we will have to circumvent this.

Someone has an idea?

**Comment by jefdaj** [comment]
Sorry don't mean to hijack the thread, but I'm excited to find somebody
else loading gitit plugins in nix! Did you have to do anything special to
get it working? I've got my fork of gitit packaged and it runs but gives an
incomprehensible (to me) error when loading any plugin:

Loading plugin 'Csv.hs'...

Network/Gitit/ContentTransformer.hs:96:0:
     error: missing binary operator before token "("
     #if MIN_VERSION_blaze_html(0,5,0)
     ^
gitit: gitit: phase `C pre-processor' failed (exitcode = 1)

Thanks
Jeff

**Comment by jraygauthier** [comment]
Are you on nix / nixos too? I never had the error you mention. However, what does the trick on my side is to run gitit in the proper environment. Is it possible that you are trying to run from a sandbox?

**Comment by jefdaj** [comment]
Yup NixOS here! It's great, or at least seems like it'll be great if it can get more volunteers going. I've been developing gitit plugins and using them for my lab wiki on Debian for a while. I have a fork of gitit (https://github.com/jefdaj/gitit), and then a set of scripts that call it with my settings, theme, etc (https://github.com/jefdaj/jeffwiki).

I tried to port both to NixOS by replacing the official gitit package in my nixpkgs, then packaging jeffwiki and having it depend on that. The plugins are loaded dynamically by putting their paths in the jeffwiki config file... or at least I hoped they would be. Gitit itself works as long as I comment them out though.

So I guess my setup is a lot like a sandbox and I should have expected that loading plugins would fail since it's trying to mix code from the gitit package with code from ~/.cabal. Next I was going to try building a monolithic package with the plugins added to Network/Gitit/Plugins/. Think that would work?
Jeff
